client.go

```diff
@@ -54,13 +63,76 @@ func (a *authCreds) RequireTransportSecurity() bool {
 	return true
 }
 
+// Open creates a new Dgraph client by parsing a connection string of the form:
+// dgraph://<optional-login>:<optional-password>@<host>:<port>?<optional-params>
+// For example `dgraph://localhost:9080?sslmode=require`
+// It connects to the gRPC endpoint and, if credentials are provided, signs in the user.
+// An error can be returned if the Dgraph cluster is not yet ready to accept requests--the text
+// of the error in this case will contain the string "Please retry".
+func Open(connStr string) (*Dgraph, error) {
+	u, err := url.Parse(connStr)
+	if err != nil {
+		return nil, fmt.Errorf("invalid connection string: %w", err)
+	}
+
+	if !strings.HasPrefix(connStr, dgraphScheme) {
+		return nil, fmt.Errorf("invalid connection string: must start with %s", dgraphScheme)
```

Comment:
we should say, invalid connection schema here

Reply:
what if I do not specify anything, can we not accept that as well? I am thinking that "localhost:8080" should work as well

Reply:
We could accept anything I guess. But localhost:8080 parses oddly. The scheme becomes the host and the host is nil...

```go
u, err := url.Parse("localhost:8080")
fmt.Println("scheme:", u.Scheme)
fmt.Println("host:", u.Host)
fmt.Println("error:", err)
```

```sh
scheme: localhost
host: 
error: <nil>
```

postgres doesn't support it. Slippery slope IMO.